File: Arrays/1431. Kids With the Greatest Number of Candies/1431. Kids With the Greatest Number of Candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function (candies, extraCandies) {
  let ans = [];
  const maxx = Math.max(...candies);
  for (const c of candies) {
    const extra = c + extraCandies;
    if (extra >= maxx) ans.push(true);
    else ans.push(false);
  }
  return ans;
};

// Time: O(n)
// Space: O(1)

File: Arrays/1431. Kids With the Greatest Number of Candies/1431. Kids With the Greatest Number of Candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        ans = []
        maxx = max(candies)
        for c in candies:
            extra = c + extraCandies
            if extra >= maxx:
                ans.append(True)
            else:
                ans.append(False)
        return ans
            
# Time: O(n)
# Space: O(1)